### ISSUE TITLE
Moment type error, although i make it with moment

### ISSUE THREAD

**Issue by huystart17** [airbnb/react-dates#1127]
Warning: Failed prop type: Invalid prop `date` of type `object` supplied to `withStyles(SingleDatePicker)`, expected `Moment`.
```javascript
import moment from "moment";
<SingleDatePick
                                            block
                                            small
                                            date={moment(gateInput.start_date)} // momentPropTypes.momentObj or null
                                            onDateChange={date =>
                                                this.onChange({ target: { name: "start_date", value: date } })
                                            } // PropTypes.func.isRequired
                                            focused={this.state.focused_start_date} // PropTypes.bool
                                            onFocusChange={({ focused }) =>
                                                this.setState({ focused_start_date: focused })
                                            } // PropTypes.func.isRequired
                                        />
```

```javascript
export class SingleDatePick extends React.Component {
    constructor(props){
        super(props);
    }
    render(){
           return (
                <SingleDatePicker
                    {...this.props}
                />
            )
    }
}
```

**Comment by Parthchokshi** [comment]
Facing same error. This happens because of initial value as null. 
Here is how to reproduce: `date={moment(null)}`.

**Comment by majapw** [comment]
@Parthchokshi has the right idea! I wonder if this is maybe something that could be addressed in https://github.com/CalebMorris/react-moment-proptypes or in moment itself. It's kind of strange that passing `null` into the moment constructor doesn't ... satisfy the proptype (i'm also surprised that `moment(null)` doesn't just resolve to `moment()` which is today's date).

**Comment by ljharb** [comment]
I would expect `moment(undefined)` and `moment()` to be the same, but i'd also expect that `moment(null)` would be an invalid date. It definitely seems like something react-moment-propTypes should be better handling.

**Comment by Parthchokshi** [comment]
True. react-moment-proptypes should give a more meaningful message.

The validation code throw an error at second level saying the date is invalid with moment constructor as null:

<img width="576" alt="screen shot 2018-06-29 at 5 00 30 pm" src="https://user-images.githubusercontent.com/6808993/42114733-22291738-7bbe-11e8-9e71-b860dd250c9d.png">
[Code](https://github.com/CalebMorris/react-moment-proptypes/blob/master/src/moment-validation-wrapper.js)

But rather throws an error saying the given prop is object and not Moment type:

<img width="735" alt="screen shot 2018-06-29 at 5 00 19 pm" src="https://user-images.githubusercontent.com/6808993/42114779-4dc50f5a-7bbe-11e8-83cb-8f70479091a4.png">
[Code](https://github.com/CalebMorris/react-moment-proptypes/blob/master/src/core.js)
